daemon/oci_windows.go

```diff
@@ -10,6 +15,72 @@ import (
 	"github.com/opencontainers/runtime-spec/specs-go"
 )
 
+func rewriteEnvVarValueForPowershell(value string) string {
+	result := ""
+	var scan scanner.Scanner
+	scan.Init(strings.NewReader(value))
+	for scan.Peek() != scanner.EOF {
+		ch := scan.Peek()
+		if ch == '"' {
+			result += "\\\""
+		} else if ch == '\\' {
+			scan.Next()
+			result += string(scan.Peek())
+		} else if ch == '$' {
+			// shell parser always write var references as ${name}
+			scan.Next()
+			ch = scan.Peek()
+			if ch != '{' {
+				result += "$"
+				result += string(ch)
+			} else {
+				name := ""
+				scan.Next()
+				for scan.Peek() != scanner.EOF && scan.Peek() != '}' {
+					name += string(scan.Peek())
+					scan.Next()
+				}
+				result += "${env:" + name + "}"
+			}
+		} else {
+			result += string(ch)
+		}
+
+		scan.Next()
+	}
+	return result
+}
+func fixupLazyEnvVars(c *container.Container, s *specs.Spec) {
+
+	if len(c.Config.LazyEnvVarNames) == 0 {
+		return
+	}
+
+	// cmd /s /c does not allow multiple statement, and there is not equivalent to `name=value echo ${name}` style syntax
+	// Tried with set name=value & echo %value% style, without success
+	// So rely on powershell to set env variables and open a nested cmd with the original args
+	toPrepend := ""
+	for _, name := range c.Config.LazyEnvVarNames {
+		upperName := strings.ToUpper(name)
+		for i, envspec := range s.Process.Env {
+			parts := strings.SplitN(envspec, "=", 2)
+			if len(parts) != 2 {
+				continue
+			}
+			envspecUpperName := strings.ToUpper(parts[0])
+			if upperName == envspecUpperName {
+				toPrepend += fmt.Sprint("${env:", name, "}=\\\"", rewriteEnvVarValueForPowershell(parts[1]), "\\\"; ")
+				s.Process.Env = append(s.Process.Env[:i], s.Process.Env[i+1:]...)
+				break
+			}
+		}
+	}
+	toPrepend += "cmd /S /C "
+	cmdLine := "'" + strings.Join(s.Process.Args, " ") + "'"
+	s.Process.Args = make([]string, 0)
+	s.Process.Args = append(s.Process.Args, "powershell", "-Command", toPrepend, "cmd", "/s", "/c", cmdLine)
```

Comment:
Do all windows containers have powershell?
Also, `append([]string{}, ...)`, or just skip the append.